examples/binaries/calc-hash/over-blocks/src/code.rs

```diff
@@ -0,0 +1,120 @@
+use crate::Method;
+use gstd::{exec, msg, ActorId, BTreeMap, Decode, Encode, MessageId, Vec};
+use shared::PackageId;
+use types::Package;
+
+#[no_mangle]
+unsafe extern "C" fn init() {
+    state::THRESHOLD = Some(msg::load().expect("Invalid threshold."));
+}
+
+#[no_mangle]
+unsafe extern "C" fn handle() {
+    unsafe {
+        let threshold = state::THRESHOLD.expect("Threshold has not been set.");
+        let method = msg::load::<Method>().expect("Invalid contract method.");
+
+        match method {
+            Method::Start { expected, id, src } => {
+                if !state::REGISTRY.contains_key(&id) {
+                    state::REGISTRY.insert(id, Package::new(expected, src));
+                }
+
+                let pkg = state::REGISTRY.get(&id).expect("Calculation not found.");
+
+                if pkg.finished() {
+                    msg::reply(pkg.result(), 0).expect("send reply failed");
+                } else {
+                    exec::wait();
+                }
+            }
+            // Proxy the `Calculate` method for mocking aggregator && calculator.
+            Method::Refuel(id) => {
+                msg::send(exec::program_id(), Method::Calculate(id), 0)
+                    .expect("Send message failed.");
+            }
+            Method::Calculate(id) => {
+                if msg::source() != exec::program_id() {
+                    panic!(
+                        "Invalid caller, this is a private method reserved for the program itself."
+                    );
+                }
+
+                let mut pkg = state::REGISTRY
```

Comment:
I don't see where you mutate `pkg`.
```suggestion
                let pkg = state::REGISTRY
```

Reply:
`pkg.calc()` is under `&mut self`